src/tool/abi/abi_writer.cpp

```diff
@@ -112,7 +112,25 @@ void writer::pop_inline_namespace()
     }
 }
 
-void writer::push_contract_guard(contract_attributes& vers)
+bool writer::push_contract_guard(version ver)
+{
+    if (std::holds_alternative<contract_version>(ver))
+    {
+        push_contract_guard(std::get<contract_version>(ver));
+        return true;
+    }
+
+    return false;
+}
+
+void writer::push_contract_guard(contract_version ver)
+{
+    contract_history history;
+    history.current_contract = ver;
+    push_contract_guard(std::move(history));
+}
+
+void writer::push_contract_guard(contract_history vers)
```

Comment:
Why is this pass-by-value. const ref seems more appropriate.

Reply:
It takes ownership